pkg/devfile/parser/data/2.0.0/components.go

```diff
@@ -131,24 +131,33 @@ func (d *Devfile200) UpdateComponent(component common.DevfileComponent) {
 	}
 }
 
-// GetCommands returns the slice of DevfileCommand objects parsed from the Devfile
-func (d *Devfile200) GetCommands() map[string]common.DevfileCommand {
+// GetCommands returns the slice of DevfileCommand objects parsed from the Devfile.
+// If the devfile has multiple commands with the same ID, an error is returned
+func (d *Devfile200) GetCommands() (map[string]common.DevfileCommand, error) {
 	commands := make(map[string]common.DevfileCommand, len(d.Commands))
 
 	for _, command := range d.Commands {
 		// we convert devfile command id to lowercase so that we can handle
 		// cases efficiently without being error prone
 		// we also convert the odo push commands from build-command and run-command flags
-		commands[command.SetIDToLower()] = command
+		commandID := command.SetIDToLower()
+		if _, exists := commands[commandID]; exists {
+			return nil, fmt.Errorf("devfile has duplicate command IDs %q", commandID)
```

Comment:
Interesting that you decided to do it here. Hmm, I was expecting it to be done in `validate` package and that wouldnt result in func signature changes 🤔

Reply:
Good question. It's because the most direct place to check for duplicate commands is when we're constructing the commands map.

If we do it elsewhere (like `validateCommands` https://github.com/openshift/odo/blob/master/pkg/devfile/validate/commands.go), the commands map has already been constructed and we won't know from that which commands (if any) are duplicated. We'd have to do a manual comparison between the list of all devfile commands and the commands Map.

Reply:
Ended up finding an approach to do the validation in the validate package instead.